DQM/BeamMonitor/plugins/BeamSpotProblemMonitor.cc

```diff
@@ -1,168 +1,130 @@
-
-  /* S.Dutta : Commenting out these variables are not used and giving error with "-Werror=unused-variable" option
-  int nbins = alarmOFFThreshold_; 
-  double hiRange = (alarmOFFThreshold_+0.5);*/
-
-  const int nvar_ = 1;
-  string coord[nvar_] = {"BeamSpotStatus"};
-  /* S.Dutta : Commenting out these variables are not used and giving error with "-Werror=unused-variable" option
-  string label[nvar_] = {"BeamSpotStatus "};*/
-
-  for (int i = 0; i < 1; i++) {
-    dbe_->setCurrentFolder(monitorName_+"FitFromScalars");
-    for (int ic=0; ic<nvar_; ++ic) {
-      TString histName(coord[ic]);
-      TString histTitle(coord[ic]);
-      string ytitle("Problem (-1)  /  OK (1)");
-      string xtitle("");
-      string options("E1");
-      bool createHisto = true;
-     switch(i){
-
-     case 0: 
-        histName += "_lumi";
-	xtitle = "Lumisection";
-
-      if (createHisto) {
-	hs[histName] = dbe_->book1D(histName,histTitle,40,0.5,40.5);
-	hs[histName]->setAxisTitle(xtitle,1);
-	hs[histName]->setAxisTitle(ytitle,2);
-
-        histName += "_all";
-        histTitle += " all";
-        hs[histName] = dbe_->book1D(histName,histTitle,40,0.5,40.5);
-        hs[histName]->getTH1()->SetCanExtend(TH1::kAllAxes);
-        hs[histName]->setAxisTitle(xtitle,1);
-        hs[histName]->setAxisTitle(ytitle,2);
-
-        }//create histo
-      }
-    }
-  }
-
-  BeamSpotError = dbe_->book1D("BeamSpotError","ERROR: Beamspot missing from scalars",20,0.5,20.5);
-  BeamSpotError->setAxisTitle("# of consecutive LSs with problem",1);
-  BeamSpotError->setAxisTitle("Problem with scalar BeamSpot",2);
-
-
-  dbe_->setCurrentFolder(monitorName_+"FitFromScalars");
-
-
-
-}
-
-//--------------------------------------------------------
-void BeamSpotProblemMonitor::beginRun(const edm::Run& r, const EventSetup& context) {
-
-
-  if (debug_) {
-    edm::LogInfo("BeamSpotProblemMonitor") << "TimeOffset = ";
-  }
-
+  iB.setCurrentFolder(monitorName_+"FitFromScalars");
+
+  const string coord{"BeamSpotStatus"};
+
+  string histName(coord + "_lumi");
+  string histTitle(coord);
+  const string ytitle("Problem (-1)  /  OK (1)");
+  const string xtitle("Lumisection");
+
+  beamSpotStatusLumi_ = iB.book1D(histName,histTitle,40,0.5,40.5);
+  beamSpotStatusLumi_->setAxisTitle(xtitle,1);
+  beamSpotStatusLumi_->setAxisTitle(ytitle,2);
+
+  histName += "_all";
+  histTitle += " all";
+  beamSpotStatusLumiAll_ = iB.book1D(histName,histTitle,40,0.5,40.5);
+  beamSpotStatusLumiAll_->getTH1()->SetCanExtend(TH1::kAllAxes);
+  beamSpotStatusLumiAll_->setAxisTitle(xtitle,1);
+  beamSpotStatusLumiAll_->setAxisTitle(ytitle,2);
+
+  //NOTE: This in principal should be a Lumi only histogram since it gets reset at every
+  // beginLuminosityBlock call. However, it is also filled at that time and the DQMStore
+  // clears all lumi histograms at postGlobalBeginLuminosityBlock!
+  beamSpotError_ = iB.book1D("BeamSpotError","ERROR: Beamspot missing from scalars",20,0.5,20.5);
```

Comment:
As the comment says, this really should have `setLumiFlag()` called on it, but that would not work because of the exact timing per lumi flags get reset.